infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -336,15 +370,25 @@ public synchronized void handleFailureDetectedMsg(CorfuPayloadMsg<FailureDetecto
             return;
         }
 
-        log.info("handleFailureDetectedMsg: Received Failures : {}",
-                msg.getPayload().getFailedNodes());
+        log.info("handleFailureDetectedMsg: Received DetectorMsg : {}", msg.getPayload());
+
         try {
+            DetectorMsg detectorMsg = msg.getPayload();
+            Layout layout = (Layout) latestLayout.clone();
+            if (!detectorMsg.getDetectorEpoch().equals(layout.getEpoch())) {
+                log.error("handleFailureDetectedMsg: Discarding stale detector message received. "
```

Comment:
Currently, this would not be possible. This is because the management client of node A would send a message to the management server of node A only. This ensures that the server sees the same or newer layout compared to the client.

If this were to occur, this message will still be rejected with the same error message.
The client would in subsequent poll rounds, would update the layout and then the server and client will be in sync again and the message will be taken care of.